src/network/core_utils.rs

```diff
@@ -292,6 +293,232 @@ impl CoreUtils {
+    /* for netavark this will be called inside container namespace. */
+    #[tokio::main]
+    async fn rename_macvlan_internal(
+        macvlan_tmp_ifname: &str,
+        macvlan_ifname: &str,
+    ) -> Result<(), std::io::Error> {
+        /* Note: most likely this is called in a seperate thread so create a new connection, rather than
+         * sharing from parent stack */
+        /* Reason: rtnetlink handle does not implements copy so we cant share it across stack and
+         * copying manually is more expesive than creating a new handle */
+        let (_connection, handle, _) = match rtnetlink::new_connection() {
+            Ok((conn, handle, messages)) => (conn, handle, messages),
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to connect: {}", err),
+                ))
+            }
+        };
+
+        tokio::spawn(_connection);
+        let mut links = handle
+            .link()
+            .get()
+            .set_name_filter(macvlan_tmp_ifname.to_string())
+            .execute();
+        match links.try_next().await {
+            Ok(Some(link)) => {
+                match handle
+                    .link()
+                    .set(link.header.index)
+                    .name(macvlan_ifname.to_string())
+                    .execute()
+                    .await
+                {
+                    Ok(_) => (),
+                    Err(err) => {
+                        return Err(std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!(
+                                "failed to rename macvlan {} to {}: {}",
+                                &macvlan_tmp_ifname, &macvlan_ifname, err
+                            ),
+                        ))
+                    }
+                }
+            }
+            Ok(None) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("macvlan interface {} not found", &macvlan_tmp_ifname),
+                ))
+            }
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to get interface {}: {}", &macvlan_tmp_ifname, err),
+                ))
+            }
+        }
+
+        // NOTE: I dont think we need to turn up the interface cause its already up.
+        // But turn up macvlan interface up again if its needed. Although i doubt.
+
+        Ok(())
+    }
+
+    /* generates macvlan and links with master veth link on the most */
+    /* moves macvlan to given network namespaces */
+    /* master veth must be up*/
+    /* by default macvlan mode is bridge*/
+    #[tokio::main]
+    pub async fn configure_macvlan_async(
+        master_veth_name: &str,
```

Comment:
The variable name is confusing, a macvlan interface does not need a veth interface. Just call it `master_name`

Reply:
changed to `master_ifname`